jiant/preprocess.py

```diff
@@ -568,40 +585,43 @@ def add_task_label_vocab(vocab, task):
         vocab.add_token_to_namespace(label, namespace)
 
 
-def add_pytorch_transformers_wpm_vocab(vocab, tokenizer_name):
-    """Add BERT/XLNet WPM vocabulary for use with pre-tokenized data.
+def add_pytorch_transformers_vocab(vocab, tokenizer_name):
+    """Add BERT/XLNet WPM or other pytorch_transformers vocabulary for use with pre-tokenized data.
 
     These tokenizers have a convert_tokens_to_ids method, but this doesn't do
     anything special, so we can just use the standard indexers.
     """
     do_lower_case = "uncased" in tokenizer_name
 
-    if tokenizer_name.startswith("bert"):
+    if tokenizer_name.startswith("bert-"):
         from pytorch_transformers import BertTokenizer
 
         tokenizer = BertTokenizer.from_pretrained(tokenizer_name, do_lower_case=do_lower_case)
-    else:
+    elif tokenizer_name.startswith("xlnet-"):
         from pytorch_transformers import XLNetTokenizer
 
         tokenizer = XLNetTokenizer.from_pretrained(tokenizer_name, do_lower_case=do_lower_case)
+    elif tokenizer_name.startswith("openai-gpt"):
+        from pytorch_transformers import OpenAIGPTTokenizer
 
-    ordered_vocab = tokenizer.convert_ids_to_tokens(range(tokenizer.vocab_size))
-    log.info("WPM vocab (%s): %d tokens", tokenizer_name, len(ordered_vocab))
-    for word in ordered_vocab:
-        vocab.add_token_to_namespace(word, tokenizer_name)
+        tokenizer = OpenAIGPTTokenizer.from_pretrained(tokenizer_name)
+    elif tokenizer_name.startswith("gpt2"):
+        from pytorch_transformers import GPT2Tokenizer
+
+        tokenizer = GPT2Tokenizer.from_pretrained(tokenizer_name)
+    elif tokenizer_name.startswith("transfo-xl-"):
+        from pytorch_transformers import TransfoXLTokenizer
 
+        tokenizer = TransfoXLTokenizer.from_pretrained(tokenizer_name)
+    elif tokenizer_name.startswith("xlm-"):
+        from pytorch_transformers import XLMTokenizer
 
-def add_openai_bpe_vocab(vocab, namespace="openai_bpe"):
-    """Add OpenAI BPE vocabulary for use with pre-tokenized data."""
-    from .openai_transformer_lm import utils as openai_utils
+        tokenizer = XLMTokenizer.from_pretrained(tokenizer_name)
 
-    id_to_wordpiece = openai_utils.reverse_encoder_dict
-    for i in range(len(id_to_wordpiece)):
-        vocab.add_token_to_namespace(id_to_wordpiece[i], namespace)
-    # Add SOS and EOS tokens to *end* of namespace, since this is where the
-    # OpenAI model expects special tokens.
-    vocab.add_token_to_namespace(utils.SOS_TOK, namespace)
-    vocab.add_token_to_namespace(utils.EOS_TOK, namespace)
+    ordered_vocab = tokenizer.convert_ids_to_tokens(range(tokenizer.vocab_size))
+    log.info("Pytorch transformer vocab (%s): %d tokens", tokenizer_name, len(ordered_vocab))
```

Comment:
=> pytorch_transformers. We should use that exact name consistently, to make it clear that we're referring to that package, not to transformers in general.